src/main/java/org/truffleruby/language/dispatch/UnresolvedDispatchNode.java

```diff
@@ -96,9 +98,9 @@ public Object executeDispatch(
                             throw new UnsupportedOperationException();
                     }
                 } else if (RubyGuards.isRubyBasicObject(receiverObject)) {
-                    newDispatchNode = doDynamicObject(frame, first, receiverObject, methodName, argumentsObjects);
+                    newDispatchNode = doDynamicObject(frame, first, receiverObject, methodName, declarationContext, argumentsObjects);
```

Comment:
Do we need to pass the DeclarationContext explicitly here or could we just read it from the frame in `lookup()`?
If the caller frame is not a Ruby frame, refinements do not apply, just like visibility.

Reply:
@eregon I think this can be read from the frame now in some of these places instead of passing explicitly.

Reply:
@eregon I've updated this to read declaration context from the frame in the latest commit